sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs

```diff
@@ -33,26 +34,25 @@ public class ServiceBusMessage
         /// </summary>
         /// <param name="body">The payload of the message as a string.</param>
         public ServiceBusMessage(string body) :
-            this(new BinaryData(body))
+            this(Encoding.UTF8.GetBytes(body))
         {
         }
 
         /// <summary>
         /// Creates a new message from the specified payload.
         /// </summary>
         /// <param name="body">The payload of the message in bytes.</param>
-        public ServiceBusMessage(ReadOnlyMemory<byte> body) :
-            this(BinaryData.FromBytes(body))
+        public ServiceBusMessage(ReadOnlyMemory<byte> body)
         {
+            AmqpMessage = new AmqpAnnotatedMessage(new ReadOnlyMemory<byte>[] { body });
         }
 
         /// <summary>
         /// Creates a new message from the specified <see cref="BinaryData"/> instance.
         /// </summary>
         /// <param name="body">The payload of the message.</param>
-        public ServiceBusMessage(BinaryData body)
+        public ServiceBusMessage(BinaryData body) : this (body?.ToMemory() ?? default)
```

Comment:
I remember we had this discussion before. At the time, BinaryData was a struct, so it could never be null. Now, it can be null. There were specific places that this made a difference, if I remember correctly. Can you make sure that's covered as a scenario?

Reply:
Yes, we have a test that passes a null value as BinaryData.
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/tests/Message/MessageTests.cs#L173

Reply:
That test doesn't actually send the message, though. I think the failures occur when the message is used, not at construction.